pkg/app/piped/apistore/eventstore/store.go

```diff
@@ -142,13 +142,16 @@ func (s *store) Getter() Getter {
 	return s
 }
 
-func (s *store) GetLatest(ctx context.Context, name string, labels map[string]string) (*model.Event, bool) {
+func (s *store) GetLatest(ctx context.Context, name string, labels map[string]string, notUsingCache bool) (event *model.Event, cacheUsed bool, ok bool) {
```

Comment:
Sorry if it's wrong.
I think `cacheUsed` indicate whether or not cache hit. Hence it always returns false when `notUsingCache` is true, 
but the other depends on  whether the cache exists.
Based on above, `cacheHit`  or something like that may be more suitable.

Reply:
In such case, this one should be `return resp.Event, true, true`?
https://github.com/pipe-cd/pipecd/pull/3108/files/5bf7e27bf20b19d118e318b3f1d463105c0e580e#diff-2a38334e3554346c2d1bce6d910c5a78b67c638bf03a7279f58c15b586c1cee5R177 (this file at L177)

Reply:
>but the other depends on whether the cache exists.

There you go 👍 

>Based on above, cacheHit or something like that may be more suitable.

Sounds better

Reply:
Alternatively, it is also fine to separate into other methods, which can share core logic.

```go
type eventGetter interface {
	GetLatest(ctx context.Context, name string, labels map[string]string) (event *model.Event, ok bool)
	GetLatestWithCache(ctx context.Context, name string, labels map[string]string) (event *model.Event, cacheHit bool, ok bool)
}
```

Reply:
One vote from me for splitting down those functions. But the returned `cacheHit` value is redundant to me. So we can have simpler things like as below, WDYT?

``` golang
type eventGetter interface {
	GetLatest(ctx context.Context, name string, labels map[string]string) (event *model.Event, ok bool)
	GetLatestFromCache(ctx context.Context, name string, labels map[string]string) (event *model.Event, ok bool)
}
```

Reply:
In the case of using just one function, since `cacheUsed` and `ok` bring the same meaning then we can have

``` golang
type eventGetter interface {
	GetLatest(ctx context.Context, name string, labels map[string]string, onlyFromCache bool) (event *model.Event, ok bool)
}
```

Reply:
I see, thanks for explaination 🙏 
In that case, I think making cache flush/update logic out of `GetLatest` maybe a better option. I mean currently, we use the GetLatest to
- Get value by key (for cache or not)
- Update cache if cache hit in case of using cache to get event status

IMO, the existing logic of GetLatest is get from cache and if there is not, fetch it and update the cache, that's simple and understandable so we should keep it as is. By add one more return value (cacheHit as you guy mentioned), and use that to control whether we should update the cache (by using cache update interface) could be more clear, wdyt

Reply:
Or another option is to make the current GetLatest interface as is, and internaly of the cache hit, refresh it and return the updated latest. In such case, the caller of event store doesn't need to care about whether event store uses cache or not.